ddp-workspace/projects/ddp-sdk/src/lib/components/address/addressEmbedded.component.ts

```diff
@@ -660,16 +677,6 @@ export class AddressEmbeddedComponent implements OnDestroy, OnInit {
     }
 
     ngOnDestroy(): void {
-        // finding that some slow http operations
-        // killed if we destroy too early
-        this.componentBusy.pipe(
-            filter(busy => !busy),
-            tap(() => {
-                this.ngUnsubscribe.next();
-                this.ngUnsubscribe.complete();
-                this.logger.logDebug(this.LOG_SOURCE, 'Unsubscribe completed.');
-            }),
-            take(1)
-        ).subscribe();
+        this.destroyComponentSignal$.next();
```

Comment:
This was another cause of some of the problem: when we move from one section to the next in the activity, ngOnDestroy gets called on the no longer visible component.
But there was a bug here and ngOnDestroy sometimes was not triggering the unsubscribe.
And this lead to unsubscribed code receiving the submitannouncement from activity component and trying to save even though the component was not visible and not fully loaded.
So refactored just to signal that ngOnDestroy has been called. Real code is above.

Reply:
Good catch!